docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+# Basic Usage
+
+## Adding Routing to an Application
+
+Begin by adding an initial route configuration, that defines a single route that maps to an identifier known as an `outlet`.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home'
+	}
+];
+```
+
+Then configure the application to be "routing" aware by registering the router with an application registry.
+
+>src/main.tsx
+```tsx
+import renderer from '@dojo/framework/widget-core/vdom';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Registry from '@dojo/framework/widget-core/Registry';
+import { registerRouterInjector } from '@dojo/framework/routing/RouterInjector';
+
+import routes from './routes';
+import App from './App';
+
+const registry = new Registry();
+// creates a router with the routes and registers the router with the registry
+registerRouterInjector(routes, registry);
+
+const r = renderer(() => <App />);
+r.mount({ registry });
+```
+
+Use the `Outlet` widget to control what widgets to render when the path for an outlet id is matched, the example will render `<div>Home</div>` for the `home` outlet.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={() => <div>Home</div>} />
+			</div>
+		);
+	}
+}
+```
+
+## Path & Query Parameters
+
+Path parameters are placeholders in the routing configuration that will match any value for the segment. The parameters are defined using curly braces, for example: `{param}`.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home'
+	}
+];
+```
+
+The parameters values are injected into to matching `Outlets`'s `renderer` property.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => <div>{`Home ${matchDetails.params.page}`}</div>} />
+			</div>
+		);
+	}
+}
+```
+
+Query parameters are also defined using curly braces split from the route path by a `?`. To define more than one query param, use the `&` delimiter.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}?{queryOne}&{queryOne}',
+		outlet: 'home'
+	}
+];
+```
+
+The query params are injected into matching `Outlet` `renderer` properties accessed from `queryParams` property.
```

Comment:
Might be worth illustrating use of `queryParams` in tandem with `params`.

Also there's a duplicate `queryOne` param in the above route - which raises another interesting question of how the framework handles such a definition. Is it an error? Or would it create an array if more than one value is given?

Reply:
That is a typo on my part, but it's a good point - currently the router doesn't have multiple query params but I can't think why we could support arrays for multiple query params.

Reply:
Agreed, this should show an example of using query params like the params above.

Reply:
Done